src/ZODB/tests/util.py

```diff
@@ -206,3 +219,117 @@ def clear_transaction_syncs():
+                now = seconds[0]
+            else:
+                now = incr()
+            return _real_gmtime(now)
+        fake_gmtime.side_effect = incr_gmtime
+
+    def __call__(self, func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            with mock.patch('time.gmtime') as fake_gmtime:
+                with mock.patch('time.time') as fake_time:
+                    self.configure_fakes(fake_gmtime, fake_time)
+                    return func(*args, **kwargs)
+        return wrapper
+
+
+def time_monotonically_increases(func_or_granularity):
+    """
+    Decorate a unittest method with this function to cause the value
+    of :func:`time.time` and :func:`time.gmtime` to monotonically
+    increase by one each time it is called. This ensures things like
+    last modified dates always increase.
+
+    We make three guarantees about the value of :func:`time.time`
+    returned while the decorated function is running:
+
+        1. It is always *at least* the value of the *real*
+           :func:`time.time`;
+
+        2. Each call returns a value greater than the previous call;
+
+        3. Those two constraints hold across different invocations of
+           functions decorated. This decorator can be applied to a
+           method in a test case::
+
+               class TestThing(unittest.TestCase)
+                   @time_monotonically_increases
+                   def test_method(self):
+                     t = time.time()
+                      ...
+
+    It can also be applied to a bare function taking any number of
+    arguments::
+
+        @time_monotonically_increases
+        def utility_function(a, b, c=1):
+           t = time.time()
+           ...
+
+    By default, the time will be incremented in 1.0 second intervals.
+    You can specify a particular granularity as an argument; this is
+    useful to keep from running too far ahead of the real clock::
+
+        @time_monotonically_increases(0.1)
+        def smaller_increment():
+            t1 = time.time()
+            t2 = time.time()
+            assrt t2 == t1 + 0.1
+    """
+    if isinstance(func_or_granularity, (six.integer_types, float)):
+        # We're being used as a factory.
+        wrapper_factory = _TimeWrapper(func_or_granularity)
+        return wrapper_factory
+
+    # We're being used bare
+    wrapper_factory = _TimeWrapper()
+    return wrapper_factory(func_or_granularity)
+
+
+def reset_monotonic_time(value=0.0):
+    """
+    Make the monotonic clock return the real time on its next
+    call.
+    """
+
+    global _current_time # pylint:disable=global-statement
+    _current_time = value
+
+
+class MonotonicallyIncreasingTimeMinimalTestLayer(MininalTestLayer):
+
+    def testSetUp(self):
+        self.fake_gmtime = mock.Mock()
+        self.fake_time = mock.Mock()
+        self.time_manager = _TimeWrapper(0.1)
+        self.time_manager.configure_fakes(self.fake_gmtime, self.fake_time)
```

Comment:
Why storing fake_gmtime, fake_time and time_manager on self ?

Reply:
Ah, that's a half-finished thought. I will adjust that as well.